### ISSUE TITLE
[Quest][GrizzlyHills]Life or Death/Kick 'EM While They are Down

### ISSUE THREAD

**Issue by TheRealestMcCoy** [dalaranwow/dalaran-wow#4527]
**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Skrimishers and the Infantry are fighting each other. This makes the quest slower.
Also some of them are spawn stacking on each other.

**Expected behaviour**: Tell us what should happen instead
The should nor be fighting.

**Steps to reproduce the problem**:
1. Go to Blue Sky Lodge.
2. 
3. 

**Include proofs for this behaviour**
Fighting
![wowscrnshot_030517_111656](https://cloud.githubusercontent.com/assets/12252441/23589089/47fcf8ca-0195-11e7-8149-36e2961f5ddf.jpg)
Spawn Stacked
![wowscrnshot_030517_111835](https://cloud.githubusercontent.com/assets/12252441/23589111/832ceba8-0195-11e7-842c-df06822875c5.jpg)

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.
[Life or Death Video](https://www.youtube.com/watch?v=6tvr3wxqCso)
[Kick 'Em While They're Down Video](https://www.youtube.com/watch?v=bLM_SPBwTWk)
**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/
[Wounded Skirimisher](http://www.wowhead.com/npc=27463/wounded-skirmisher)
[Wounded Westfall Infantry](http://www.wowhead.com/npc=27482/wounded-westfall-infantry)

**Include Screenshots from the issue if necessary**

**Comment by Teridan-dal-wow** [comment]
Confirmed. These double/tripple spawns needs to be rechecked. A quick fix for the combat state:

``` sql
-- Wounded Westfall Infantry SAI
SET @ENTRY := 27482;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,8,0,100,512,0,0,0,0,22,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - On Spellhit - Set Event Phase 1"),
(@ENTRY,0,1,0,1,1,100,0,0,0,3000,3000,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - Out of Combat - Say Line 0"),
(@ENTRY,0,2,3,1,1,100,512,2000,2000,2000,2000,53,1,27482,0,12296,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - Out of Combat - Start Waypoint (Phase 1)"),
(@ENTRY,0,3,4,61,1,100,512,0,0,0,0,18,128,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - Out of Combat - Set Flag Not Attackable (Phase 1)"),
(@ENTRY,0,4,5,61,1,100,512,0,0,0,0,11,48813,0,0,0,0,0,7,0,0,0,0,0,0,0,"Wounded Westfall Infantry - Out of Combat - Cast 'Kill Credit' (Phase 1)"),
(@ENTRY,0,5,0,61,1,100,512,0,0,0,0,22,2,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - Out of Combat - Set Event Phase 2 (Phase 1)"),
(@ENTRY,0,6,0,40,2,100,512,2,27482,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - On Waypoint 2 Reached - Despawn Instant (Phase 1)"),
(@ENTRY,0,7,0,25,0,100,0,0,0,0,0,8,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Wounded Westfall Infantry - On Reset - Set Reactstate Defensive");
```
However for the horde units there is a issue, they have `ScriptName=npc_wounded_skirmisher` and i can't change it.
The ally units will attack only if attacked now.

**Comment by Rushor** [comment]
it's intended that they should fight each other but they need the flagextra, that the creatures should take no damage if they attack each other. Only player and pets should cause real damage to them

**Comment by Rushor** [comment]
i'm going to covert the corescript then

**Comment by Teridan-dal-wow** [comment]
@TheRealestMcCoy  Thanks for the report!

What is done here:
-added the missing spawns
-set emotes for every creature
-removed the wrong spawns

To do:
-set reach = defensive to entry 27482 and 27463
-add spawn behavior (There are ~ 15 horde creatures and around ~ 15 alliance units all together, however only half of them are spawned at the same time (7-8 horde and 7-8 ally) and as soon as 1 of them is killed there is an instant spawn on other spawn location (from same faction)).
Note: I am not sure if this is wrath behavior or added in later exp. Also if this instant spawn in implemented maybe remove the exp reward from them as it will be easily farmable spot. Couldn't find any wowhead comments relevent to this.

How it looks like now:
![image](https://cloud.githubusercontent.com/assets/6797611/23646945/f5098470-031b-11e7-83f8-7cf9500c81c5.png)
![image](https://cloud.githubusercontent.com/assets/6797611/23646949/fce31328-031b-11e7-8256-e8192d267100.png)
![image](https://cloud.githubusercontent.com/assets/6797611/23646957/07c156a6-031c-11e7-969d-a9a15401683b.png)

``` sql
SET @cguid:=5005372;
DELETE FROM `creature` WHERE `guid` BETWEEN @CGUID+0 AND @CGUID+11;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+0, 27463, 571, 0, 0, 1, 1, 0, 0, 4203.006, -3047.501, 281.2304, 4.415683, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+1, 27482, 571, 0, 0, 1, 1, 0, 0, 4184.69, -2980.583, 284.83, 3.543018, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27482 (Area: -1)
(@CGUID+2, 27457, 571, 0, 0, 1, 1, 0, 0, 4137.022, -3007.063, 285.8553, 5.51524, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27457 (Area: -1) (Auras: 29266 - 29266)
(@CGUID+3, 27481, 571, 0, 0, 1, 1, 0, 0, 4141.247, -3012.803, 285.4366, 0.2443461, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27481 (Area: -1) (Auras: 29266 - 29266)
(@CGUID+4, 27463, 571, 0, 0, 1, 1, 0, 0, 4166.682, -2964.984, 283.2789, 2.740167, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+5, 27482, 571, 0, 0, 1, 1, 0, 0, 4206.615, -3002.458, 282.8639, 3.176499, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27482 (Area: -1)
(@CGUID+6, 27482, 571, 0, 0, 1, 1, 0, 0, 4180.862, -3014.165, 280.0445, 5.497787, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27482 (Area: -1)
(@CGUID+7, 27463, 571, 0, 0, 1, 1, 0, 0, 4139.789, -3021.132, 285.5673, 5.899213, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+8, 27482, 571, 0, 0, 1, 1, 0, 0, 4147.580, -2955.251, 283.479, 5.757801, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+9, 27482, 571, 0, 0, 1, 1, 0, 0, 4140.755, -3006.965, 285.378, 4.524723, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+10, 27482, 571, 0, 0, 1, 1, 0, 0, 4201.036, -2985.048, 283.169, 0.860837, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420), -- 27463 (Area: -1)
(@CGUID+11, 27482, 571, 0, 0, 1, 1, 0, 0, 4128.313, -2937.812, 281.963, 0.605581, 300, 0, 0, 1, 0, 0, 0, 0, 0, 23420); -- 27463 (Area: -1)

DELETE FROM `creature_addon` WHERE `guid` BETWEEN @CGUID+0 AND @CGUID+11;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 102453 AND 102457;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 102459 AND 102463;
DELETE FROM `creature_addon` WHERE `guid` IN (102466, 102467, 102469);
DELETE FROM `creature_addon` WHERE `guid` IN (104181, 104182, 104184, 104185);
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 104187 AND 104189;

DELETE FROM `creature_addon` WHERE `guid` BETWEEN 102034 AND 102038;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 102041 AND 102047;

DELETE FROM `creature_addon` WHERE `guid` BETWEEN 103956 AND 103960;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 103972 AND 103978;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 103984 AND 103987;


INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `auras`) VALUES
(102034, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102035, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102036, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102037, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102038, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102041, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102042, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102043, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102044, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102045, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102046, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(102047, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103956, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103957, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103958, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103959, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103960, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103972, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103973, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103974, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103975, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103976, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103977, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103978, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103984, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103985, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103986, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(103987, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(@CGUID+0, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+1, 0, 0, 8, 257, 0, ''), -- 27482
(@CGUID+2, 0, 0, 0, 1, 0, '29266'), -- 27457 - 29266 - 29266
(@CGUID+3, 0, 0, 0, 1, 0, '29266'), -- 27481 - 29266 - 29266
(@CGUID+4, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+5, 0, 0, 8, 257, 0, ''), -- 27482
(@CGUID+6, 0, 0, 8, 257, 0, ''), -- 27482
(@CGUID+7, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+8, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+9, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+10, 0, 0, 8, 257, 0, ''), -- 27463
(@CGUID+11, 0, 0, 8, 257, 0, ''), -- 27463
(102453, 0, 0, 8, 257, 0, ''), -- 27463
(102454, 0, 0, 8, 257, 0, ''), -- 27463
(102455, 0, 0, 8, 257, 0, ''), -- 27463
(102456, 0, 0, 8, 257, 0, ''), -- 27463
(102457, 0, 0, 8, 257, 0, ''), -- 27463
(102459, 0, 0, 8, 257, 0, ''), -- 27463
(102460, 0, 0, 8, 257, 0, ''), -- 27463
(102461, 0, 0, 8, 257, 0, ''), -- 27463
(102462, 0, 0, 8, 257, 0, ''), -- 27463
(102463, 0, 0, 8, 257, 0, ''), -- 27463
(102466, 0, 0, 8, 257, 0, ''), -- 27463
(102467, 0, 0, 8, 257, 0, ''), -- 27463
(102469, 0, 0, 8, 257, 0, ''), -- 27463
(104181, 0, 0, 8, 257, 0, ''), -- 27463
(104182, 0, 0, 8, 257, 0, ''), -- 27463
(104184, 0, 0, 8, 257, 0, ''), -- 27463
(104185, 0, 0, 8, 257, 0, ''), -- 27463
(104187, 0, 0, 8, 257, 0, ''), -- 27463
(104188, 0, 0, 8, 257, 0, ''), -- 27463
(104189, 0, 0, 8, 257, 0, '');


DELETE FROM `creature` WHERE `guid` IN (104190, 102458, 102464);
DELETE FROM `creature_addon` WHERE `guid` IN (104190, 102458, 102464);


UPDATE `creature` SET `position_x` = 4018.115 , `position_y` = -2835.341 , `position_z` = 279.326 , `orientation` = 1.795531 WHERE `guid` = 102466; 
UPDATE `creature` SET `position_x` = 4133.794 , `position_y` = -2896.201 , `position_z` = 279.241 , `orientation` = 5.215924 WHERE `guid` = 102457; 
UPDATE `creature` SET `orientation` = 3.488043 WHERE `guid` = 102462; 
UPDATE `creature` SET `orientation` = 2.486658 WHERE `guid` = 102464; 
UPDATE `creature` SET `position_x` = 4190.735 , `position_y` = -2958.304 , `position_z` = 283.319 , `orientation` = 0.719515 WHERE `guid` = 102463; 
UPDATE `creature` SET `position_x` = 4172.601 , `position_y` = -2906.019 , `position_z` = 282.077 , `orientation` = 4.351964 , `spawndist` = 0 , `MovementType` = 0 WHERE `guid` = 104181; 
```

**Comment by Rushor** [comment]
also: the combat abilities are missing:

http://www.wowhead.com/npc=27482/wounded-westfall-infantry#comments

By lothandar (231 – 3) on 2010/01/25 (Patch 3.3.0)		
Good to know:

These are actually tougher and have more annoying abilities than regular westfall infantry, so if you're doing the dailies, leave these alone and go for the healthy ones. They're weaker. (Blizzard did the funny here to discourage people from killing wounded enemies on the battlefield, clever, I admit!)

By sluggolito (1,872 – 10·34) on 2010/06/15 (Patch 3.3.3)		
These are no normal Infantry. These are Paladins!
Saw one today use Holy Shock and Seal of Command.
Last edited by sluggolito on 2010/06/15 (Patch 3.3.3)

Same for the horde ones

**Comment by Rushor** [comment]
DB/Quest: Life or Death/Kick 'EM While They are Down - Wounded Skirmisher +Wounded Westfall Infantry

* add missing spawns and kneel emotes
* move Wounded Skirmisher script to DB only
* add combat abilities ('Cleave', 'Hamstring', 'Mortal Strike' for Wounded Skirmisher)('Seal of Command'  and 'Holy Shock' for Wounded Westfall Infantry)
* set reactstate defensive on Reset
* add pooling for spawnlocations of Wounded Skirmisher and Wounded Westfall Infantry
* closes: https://github.com/dalaranwow/dalaran-wow/issues/4527
* commit: 06717bb4c6de6a1e21adea723d04500f4e64fac4

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.


Thanks for all the help.
